data-semantics.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: page
+title: Data Semantics Guidelines
+---
+<div id="toc"></div>
+
+## Introduction
+
+OpenTracing defines an API through which application instrumentation can log data to a pluggable tracer. In general, there is no guarantee made by OpenTracing about the way that data will be handled by an underlying tracer. That said, the APIs are by design open-ended: what type of data should be provided to them?
+
+A general understanding between the instrumentor and tracer developers adds great value: if certain known tag key/values are used for common application scenarios, tracers can choose to pay special attention to them. The same is true of `log`ged events, and span structure in general.
+
+As an example, consider the common case of a HTTP-based application server. The URL of an incoming request that the application is handling is often useful for diagnostics, as well as the HTTP verb and the resultant status code. An instrumentor could choose to report the URL in a tag named `URL`, or perhaps named `http.url`--either would be valid from the pure API perspective. But if the Tracer wishes to add intelligence, such as indexing on the URL value or sampling proactively for requests to a particular endpoint, it must know where to look for relevant data. In short, when tag names and other instrumentor-provided values are used consistently, the tracers on the other side of the API can employ more intelligence.
+
+The guidelines contained describe a common ground on which instrumentors and tracer authors can build beyond pure data collection. Adherence to the guidelines is optional but highly recommended for instrumetors.
+
+
+
+## Spans
+
+
+### Span Naming
+
+Spans carry tags, logs, and attributes, but they also have a top-level **operation name**. This should be a low-cardinality string value representing the type of work being done in the span. Typically, a process, framework, library, or module name is a good guideline.
+
+For instance, a request being handled by a Flask-based application might be instrumented using a WSGI middleware. In this situation, the first span operation name might be `WSGI`. If the application subsequently makes a database query, the span name might be `SQL`.
```

Comment:
Summarizing the above, I think we're into `component` as a thing, and "package" feels like a decent abstraction to capture it.  The last decision is how formalized it becomes vs `operation_name`.

It seems as though there's 4 categories of option:

1. Demote `operation_name` and strongly suggest both it and `component`
  - I don't think we want to remove all mandatory span labeling, so this is out.

2. Elevate `component` to equivalent (`span.setComponentName` and inclusion in constructors)
  - This is tempting to me.  Would it be considered too onerous?

3. Replace `operation_name` with `component` and strongly suggest `operation_name` as a tag
  - This is the direction AppNeta went, though it's the opposite of Zipkin I believe?  Potentially controversial.

4. Strongly suggest `component` as a tag
  - I suspect this will end up with weaker compliance than we want.

My vote is 2 or 3.  Thoughts?